pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go

```diff
@@ -209,6 +209,9 @@ func (r *KsvcReconciler) Reconcile() (*knservingv1.ServiceStatus, error) {
 	existing.Spec.ConfigurationSpec = desired.Spec.ConfigurationSpec
 	existing.ObjectMeta.Labels = desired.ObjectMeta.Labels
 	existing.Spec.Traffic = desired.Spec.Traffic
+	if value, ok := desired.ObjectMeta.Annotations[constants.RollOutDurationAnnotationKey]; ok {
```

Comment:
@henrysecond1 Is this the only field at the metadata annotation level?

Reply:
We might want to implement it in a more generic way to reconcile the annotations under knative serving api group, so we do not need handle this one off case.

Reply:
> @henrysecond1 Is this the only field at the metadata annotation level?

@yuzisun AFAIK, yes it's the only field at ksvc metadata annotation level (https://github.com/kserve/kserve/blob/v0.10.1/pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go#L97-L103)

> We might want to implement it in a more generic way to reconcile the annotations under knative serving api group, so we do not need handle this one off case.

I understand, but when I tried to reconcile whole annotations, knative does not allow it with validation errors `"validation.webhook.serving.knative.dev" denied the request: validation failed: annotation value is immutable: metadata.annotations.serving.knative.dev/creator`

or we can maintain set of annotations managed by kserve like

```go
var managedKsvcAnnotations = map[string]bool{
	constants.RollOutDurationAnnotationKey: true,
}

...

	for annotationKey := range managedKsvcAnnotations {
		if value, ok := desired.ObjectMeta.Annotations[annotationKey]; ok {
			existing.ObjectMeta.Annotations[annotationKey] = value
		}
	}
```

I think it's a more generic way. what do you think about it?

Reply:
@henrysecond1 sounds good, we need to reconcile this explicit list of ksvc annotations.

Reply:
@yuzisun added a new commit for it. PTAL